### PR TITLE
ec2 names & AutoScalingRollingUpdate

### PR DESCRIPTION
**Pull request by Eugst**

### CODE REVIEW

**Comment by andreaswittig** [comment]
Hi @Eugst, thanks for your PR. Could you please explain this PR?

File: template.json [hunk 1]
```diff
@@ -284,7 +334,7 @@
               }
             },
             "sources": {
-              "/opt": "https://github.com/widdix/aws-s3-virusscan/archive/master.zip"
+              "/opt": "https://github.com/Eugst/aws-s3-virusscan/archive/master.zip"
```
Comment (andreaswittig) [inline template.json]: Should not be changed!

**Comment by Eugst** [comment]
Hi @andreaswittig!
hehe, it's already out of date PR...
But let me explain idea, and if you would like it - I will update with last changes.
Imagine that we have just one ec2 instance. During ClamAv scanning file, other files that in queue could be deleted from S3. This situation make me annoying to Clean DLQ manually several times per day..
So, I decided it's fair enough, if after file came to DLQ it will be checked if it's present on S3, if not - removed from DLQ.

Let me know if you have concerns about, 
if it's totally wrong - please explain and close this PR.
if it's need polishing like source - just let me know I will fix it

**Comment by andreaswittig** [comment]
Hi,

thanks for explaining your PR. I got your problem. An easier and more efficient way of solving this problem might be to handle the „object not found“ error in the CalmAV worker to avoid retries and create messages for the DQL at all. Getting messages from the DQL and requesting S3 objects causes unneeded requests causing unnecessary costs.

What do you think? Any other ideas?

Andreas

**Comment by michaelwittig** [comment]
Maybe we should go back to https://github.com/widdix/aws-s3-virusscan/pull/1

**Comment by Eugst** [comment]
now we have chicken-egg issue :)
It was my first idea if object not present - then just throw it from queue.

As for me - both solutions are good, and have chance to live :)
What do you prefer?

**Comment by michaelwittig** [comment]
After considering the problem, I think that your initial idea is the better solution. I opened a PR with your initial change here: https://github.com/widdix/aws-s3-virusscan/pull/5

This only works properly für newly created objects because there we hav read-after-write consistency. It does not work properly for changed objects. THis problem can be solved by enabling versioning.

**Pull request closed without merging**